x-pack/legacy/plugins/maps/public/layers/sources/client_file_source/constants.js

```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const DEFAULT_FILTER_BY_MAP_BOUNDS = false;
+export const DEFAULT_APPLY_GLOBAL_QUERY = false;
```

Comment:
agreed that `DEFAULT_APPLY_GLOBAL_QUERY` can be false always.

I don't think `DEFAULT_FILTER_BY_MAP_BOUNDS` should be hardcoded to false. I would just use the exact same logic we have in the es-search-source wizard.

https://github.com/thomasneirynck/kibana/blob/367da246985e311687b5c3af08c1538a9e9a26cc/x-pack/legacy/plugins/maps/public/layers/sources/es_search_source/create_source_editor.js#L92

```indexHasSmallDocCount = indexDocCount <= ES_SIZE_LIMIT;```

It will actually be easier here, since you don't need to do a ES-count API call. We already know exactly how many features we have. It's just the `features.length` of the feature-collection. If it's over the ES_SIZE_LIMIT, set this to false.